src/screens/NewTerm.js

```diff
@@ -0,0 +1,201 @@
+  const findSelectedId = Text => {
+    setSelectedCategory(Text);
+    const selectedCat = categories.find(cat => cat.name.en === Text);
+    setCategoryId(selectedCat.id);
+  };
+  const addnewPhraseForCategory = () => {
+    const newPhrase = {
+      catId: categoryId,
+      id: uuid(),
+      name: {en: englishText, mg: malagasyText},
+    };
+    addNewPhrases(newPhrase);
+    // clear input
+    setMalagasyText('');
+    setEnglishText('');
+  };
+
+  useEffect(() => {
+    // fetch categories
+    const categories = getAllCategories();
+    setCategories(categories);
+  }, []);
+
+  return (
+    <SafeAreaView style={{flex: 1}}>
+      <KeyboardAvoidingView style={{flex: 1}} behavior="padding">
+        <View style={{paddingHorizontal: 35, paddingVertical: 23}}>
+          <View style={styles.header}>
+            <ToolBar
+              button={
+                <ToolButton
+                  onPress={() => {
+                    navigation.navigate('Home');
+                  }}>
+                  <BackIcon width={24} height={24} fill="#FFFFFF" />
+                </ToolButton>
+              }
+            />
+            <ToolBar
+              button={
+                <LanguageSwitcher
+                  firstLanguage={LANGUAGE_NAMES.EN}
+                  LeftText="EN"
+                  RightText="MA"
+                  color="#FFFFFF"
+                  iconType=""
+                  iconName="swap-horiz"
+                  onPress={() => null}
+                  iconSize={24}
+                />
+              }
+            />
+            <ToolBar
+              button={
+                <ToolButton onPress={action('clicked-add-button')}>
+                  <ModeIcon width={24} height={24} fill="#FFFFFF" />
+                </ToolButton>
+              }
+            />
+          </View>
+          <View style={styles.headerCategoryWrapper}>
+            <SectionHeading text="Category: " />
+            <View>
+              <Picker
+                selectedValue={selectedCategory}
+                style={{
+                  width: 300,
+                  height: 100,
+                }}
+                mode="dropdown"
+                onValueChange={Text => findSelectedId(Text)}>
+                <Picker.Item
+                  label={'Select Category'}
+                  value={null}
+                  color="#06B6D4"
+                />
+                {categories.map(categoryName => (
+                  <Picker.Item
+                    label={categoryName.name.en}
+                    key={categoryName.id}
+                    value={categoryName.name.en}
+                  />
+                ))}
+              </Picker>
+            </View>
+          </View>
+          <View style={styles.headerPhrases}>
+            <Text style={{paddingBottom: 15}}>
+              <SectionHeading text="The phrase in English: " />
+            </Text>
+            <Textarea
+              phrase={englishText}
```

Comment:
Hey, it would be better if the name of the this value in line 129 should be improved and more descriptive. It should describe what kind of text it is.